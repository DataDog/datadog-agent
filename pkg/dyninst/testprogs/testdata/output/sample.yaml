stackC: ""
testArrayOfArrays: '{"stack_frames":""}'
testArrayOfArraysOfArrays: '{"stack_frames":""}'
testArrayOfStrings: '{"stack_frames":""}'
testArrayOfStructs: '{"stack_frames":""}'
testBigStruct: ""
testBoolArray: '{"stack_frames":""}'
testByteArray: '{"stack_frames":""}'
testChannel: ""
testCombinedByte: '{"stack_frames":"","captures":{"x":3,"y":0,"w":2}}'
testEmptySlice: '{"stack_frames":""}'
testEmptySliceOfStructs: '{"stack_frames":""}'
testEmptyStruct: '{"captures":{"e":{}},"stack_frames":""}'
testError: ""
testInt8Array: '{"stack_frames":""}'
testInt16Array: '{"stack_frames":""}'
testInt32Array: '{"stack_frames":""}'
testInt64Array: '{"stack_frames":""}'
testIntArray: '{"stack_frames":""}'
testInterface: '{"stack_frames":""}'
testLinkedList: ""
testMapStringToInt: '{"stack_frames":""}'
testMultipleSimpleParams: '{"stack_frames":"","captures":{"a":false,"b":42,"c":122,"d":1337}}'
testNilPointer: '{"captures":{"z":"nil","a":1},"stack_frames":""}'
testNilSlice: '{"stack_frames":""}'
testNilSliceOfStructs: '{"stack_frames":""}'
testNilSliceWithOtherParams: '{"stack_frames":"","captures":{"a":1}}'
testOneStringInStructPointer: '{"stack_frames":""}'
testPointerLoop: ""
testRuneArray: '{"stack_frames":""}'
testSingleBool: '{"stack_frames":"","captures":{"x":true}}'
testSingleByte: '{"stack_frames":"","captures":{"x":97}}'
testSingleFloat32: '{"stack_frames":"","captures":{"x":0}}'
testSingleFloat64: '{"stack_frames":"","captures":{"x":0}}'
testSingleInt: '{"stack_frames":"","captures":{"x":-1512}}'
testSingleInt8: '{"stack_frames":"","captures":{"x":-8}}'
testSingleInt16: '{"stack_frames":"","captures":{"x":-16}}'
testSingleInt32: '{"stack_frames":"","captures":{"x":-32}}'
testSingleInt64: '{"stack_frames":"","captures":{"x":-64}}'
testSingleRune: '{"stack_frames":"","captures":{"x":1}}'
testSingleString: '{"stack_frames":""}'
testSingleUint: '{"captures":{"x":1512},"stack_frames":""}'
testSingleUint8: '{"stack_frames":"","captures":{"x":8}}'
testSingleUint16: '{"stack_frames":"","captures":{"x":16}}'
testSingleUint32: '{"stack_frames":"","captures":{"x":32}}'
testSingleUint64: '{"captures":{"x":64},"stack_frames":""}'
testSliceOfSlices: '{"stack_frames":""}'
testStringArray: '{"stack_frames":""}'
testStringPointer: '{"stack_frames":""}'
testStringSlice: '{"stack_frames":""}'
testStruct: '{"stack_frames":""}'
testStructSlice: '{"stack_frames":""}'
testThreeStrings: '{"stack_frames":""}'
testThreeStringsInStruct: '{"stack_frames":""}'
testThreeStringsInStructPointer: '{"stack_frames":""}'
testTypeAlias: '{"stack_frames":"","captures":{"x":3}}'
testUint8Array: '{"stack_frames":""}'
testUint16Array: '{"stack_frames":""}'
testUint32Array: '{"stack_frames":""}'
testUint64Array: '{"stack_frames":""}'
testUintArray: '{"stack_frames":""}'
testUintPointer: '{"stack_frames":""}'
testUintSlice: '{"stack_frames":""}'
