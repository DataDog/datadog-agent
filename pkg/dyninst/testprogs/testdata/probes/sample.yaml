binary: sample
probes:
- id: testSingleByte
  type: LOG_PROBE
  where:
    methodName: main.testSingleByte
  captureSnapshot: true
- id: testSingleRune
  type: LOG_PROBE
  where:
    methodName: main.testSingleRune
  captureSnapshot: true
- id: testSingleBool
  type: LOG_PROBE
  where:
    methodName: main.testSingleBool
  captureSnapshot: true
- id: testSingleInt
  type: LOG_PROBE
  where:
    methodName: main.testSingleInt
  captureSnapshot: true
- id: testSingleInt8
  type: LOG_PROBE
  where:
    methodName: main.testSingleInt8
  captureSnapshot: true
- id: testSingleInt16
  type: LOG_PROBE
  where:
    methodName: main.testSingleInt16
  captureSnapshot: true
- id: testSingleInt32
  type: LOG_PROBE
  where:
    methodName: main.testSingleInt32
  captureSnapshot: true
- id: testSingleInt64
  type: LOG_PROBE
  where:
    methodName: main.testSingleInt64
  captureSnapshot: true
- id: testSingleUint
  type: LOG_PROBE
  where:
    methodName: main.testSingleUint
  captureSnapshot: true
- id: testSingleUint8
  type: LOG_PROBE
  where:
    methodName: main.testSingleUint8
  captureSnapshot: true
- id: testSingleUint16
  type: LOG_PROBE
  where:
    methodName: main.testSingleUint16
  captureSnapshot: true
- id: testSingleUint32
  type: LOG_PROBE
  where:
    methodName: main.testSingleUint32
  captureSnapshot: true
- id: testSingleUint64
  type: LOG_PROBE
  where:
    methodName: main.testSingleUint64
  captureSnapshot: true
- id: testSingleFloat32
  type: LOG_PROBE
  where:
    methodName: main.testSingleFloat32
  captureSnapshot: true
- id: testSingleFloat64
  type: LOG_PROBE
  where:
    methodName: main.testSingleFloat64
  captureSnapshot: true
- id: testTypeAlias
  type: LOG_PROBE
  where:
    methodName: main.testTypeAlias
  captureSnapshot: true
- id: testSingleString
  type: LOG_PROBE
  where:
    methodName: main.testSingleString
  captureSnapshot: true
- id: testThreeStrings
  type: LOG_PROBE
  where:
    methodName: main.testThreeStrings
  captureSnapshot: true
- id: testThreeStringsInStruct
  type: LOG_PROBE
  where:
    methodName: main.testThreeStringsInStruct
  captureSnapshot: true
- id: testThreeStringsInStructPointer
  type: LOG_PROBE
  where:
    methodName: main.testThreeStringsInStructPointer
  captureSnapshot: true
- id: testOneStringInStructPointer
  type: LOG_PROBE
  where:
    methodName: main.testOneStringInStructPointer
  captureSnapshot: true
- id: testByteArray
  type: LOG_PROBE
  where:
    methodName: main.testByteArray
  captureSnapshot: true
- id: testRuneArray
  type: LOG_PROBE
  where:
    methodName: main.testRuneArray
  captureSnapshot: true
- id: testStringArray
  type: LOG_PROBE
  where:
    methodName: main.testStringArray
  captureSnapshot: true
- id: testBoolArray
  type: LOG_PROBE
  where:
    methodName: main.testBoolArray
  captureSnapshot: true
- id: testIntArray
  type: LOG_PROBE
  where:
    methodName: main.testIntArray
  captureSnapshot: true
- id: testInt8Array
  type: LOG_PROBE
  where:
    methodName: main.testInt8Array
  captureSnapshot: true
- id: testInt16Array
  type: LOG_PROBE
  where:
    methodName: main.testInt16Array
  captureSnapshot: true
- id: testInt32Array
  type: LOG_PROBE
  where:
    methodName: main.testInt32Array
  captureSnapshot: true
- id: testInt64Array
  type: LOG_PROBE
  where:
    methodName: main.testInt64Array
  captureSnapshot: true
- id: testUintArray
  type: LOG_PROBE
  where:
    methodName: main.testUintArray
  captureSnapshot: true
- id: testUint8Array
  type: LOG_PROBE
  where:
    methodName: main.testUint8Array
  captureSnapshot: true
- id: testUint16Array
  type: LOG_PROBE
  where:
    methodName: main.testUint16Array
  captureSnapshot: true
- id: testUint32Array
  type: LOG_PROBE
  where:
    methodName: main.testUint32Array
  captureSnapshot: true
- id: testUint64Array
  type: LOG_PROBE
  where:
    methodName: main.testUint64Array
  captureSnapshot: true
- id: testArrayOfArrays
  type: LOG_PROBE
  where:
    methodName: main.testArrayOfArrays
  captureSnapshot: true
- id: testArrayOfStrings
  type: LOG_PROBE
  where:
    methodName: main.testArrayOfStrings
  captureSnapshot: true
- id: testArrayOfArraysOfArrays
  type: LOG_PROBE
  where:
     methodName: main.testArrayOfArraysOfArrays
  captureSnapshot: true
# - id: testArrayOfStructs
#   type: LOG_PROBE
#   where:
#     methodName: main.testArrayOfStructs
#   captureSnapshot: true
- id: testUintSlice
  type: LOG_PROBE
  where:
    methodName: main.testUintSlice
  captureSnapshot: true
- id: testEmptySlice
  type: LOG_PROBE
  where:
    methodName: main.testEmptySlice
  captureSnapshot: true
- id: testSliceOfSlices
  type: LOG_PROBE
  where:
    methodName: main.testSliceOfSlices
  captureSnapshot: true
- id: testStructSlice
  type: LOG_PROBE
  where:
    methodName: main.testStructSlice
  captureSnapshot: true
- id: testEmptySliceOfStructs
  type: LOG_PROBE
  where:
    methodName: main.testEmptySliceOfStructs
  captureSnapshot: true
- id: testNilSliceOfStructs
  type: LOG_PROBE
  where:
    methodName: main.testNilSliceOfStructs
  captureSnapshot: true
- id: testStringSlice
  type: LOG_PROBE
  where:
    methodName: main.testStringSlice
  captureSnapshot: true
- id: testNilSliceWithOtherParams
  type: LOG_PROBE
  where:
    methodName: main.testNilSliceWithOtherParams
  captureSnapshot: true
- id: testNilSlice
  type: LOG_PROBE
  where:
    methodName: main.testNilSlice
  captureSnapshot: true
- id: testStruct
  type: LOG_PROBE
  where:
    methodName: main.testStruct
  captureSnapshot: true
- id: testEmptyStruct
  type: LOG_PROBE
  where:
    methodName: main.testEmptyStruct
  captureSnapshot: true
- id: testUintPointer
  type: LOG_PROBE
  where:
    methodName: main.testUintPointer
  captureSnapshot: true
- id: testStringPointer
  type: LOG_PROBE
  where:
    methodName: main.testStringPointer
  captureSnapshot: true
- id: testNilPointer
  type: LOG_PROBE
  where:
    methodName: main.testNilPointer
  captureSnapshot: true
- id: testCombinedByte
  type: LOG_PROBE
  where:
    methodName: main.testCombinedByte
  captureSnapshot: true
- id: testMultipleSimpleParams
  type: LOG_PROBE
  where:
    methodName: main.testMultipleSimpleParams
  captureSnapshot: true
- id: testMapStringToInt
  type: LOG_PROBE
  where:
    methodName: main.testMapStringToInt
  captureSnapshot: true
- id: testInterface
  type: LOG_PROBE
  where:
    methodName: main.testInterface
  captureSnapshot: true
- id: testError
  type: LOG_PROBE
  where:
    methodName: main.testError
  captureSnapshot: true
- id: testBigStruct
  type: LOG_PROBE
  where:
    methodName: main.testBigStruct
  captureSnapshot: true
- id: stackC
  type: LOG_PROBE
  where:
    methodName: main.stackC
  captureSnapshot: true
- id: testChannel
  type: LOG_PROBE
  where:
    methodName: main.testChannel
  captureSnapshot: true
- id: testLinkedList
  type: LOG_PROBE
  where:
    methodName: main.testLinkedList
  captureSnapshot: true
- id: testPointerLoop
  type: LOG_PROBE
  where:
    methodName: main.testPointerLoop
  captureSnapshot: true
