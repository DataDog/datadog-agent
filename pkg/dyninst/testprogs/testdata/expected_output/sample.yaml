main.stack_C: ""
main.test_array_of_arrays: ""
main.test_array_of_arrays_of_arrays: ""
main.test_array_of_strings: ""
main.test_array_of_structs: ""
main.test_big_struct: ""
main.test_bool_array: ""
main.test_byte_array: ""
main.test_channel: ""
main.test_combined_byte: ""
main.test_empty_slice: ""
main.test_empty_slice_of_structs: ""
main.test_empty_struct: ""
main.test_error: ""
main.test_int_array: ""
main.test_int8_array: ""
main.test_int16_array: ""
main.test_int32_array: ""
main.test_int64_array: ""
main.test_interface: ""
main.test_linked_list: ""
main.test_map_string_to_int: ""
main.test_multiple_simple_params: ""
main.test_nil_pointer: ""
main.test_nil_slice: ""
main.test_nil_slice_of_structs: ""
main.test_nil_slice_with_other_params: ""
main.test_one_string_in_struct_pointer: ""
main.test_pointer_loop: ""
main.test_rune_array: ""
main.test_single_bool: |
    {"stack_frames":[670432,672224,298500,517188],"captures":{"x":true}}
main.test_single_byte: |
    {"stack_frames":[670400,672224,298500,517188],"captures":{"x":97}}
main.test_single_float32: |
    {"stack_frames":[670608,672224,298500,517188],"captures":{"x":0}}
main.test_single_float64: |
    {"stack_frames":[670624,672224,298500,517188],"captures":{"x":0}}
main.test_single_int: |
    {"stack_frames":[670448,672224,298500,517188],"captures":{"x":-1512}}
main.test_single_int8: |
    {"stack_frames":[670464,672224,298500,517188],"captures":{"x":-8}}
main.test_single_int16: |
    {"stack_frames":[670480,672224,298500,517188],"captures":{"x":-16}}
main.test_single_int32: |
    {"stack_frames":[670496,672224,298500,517188],"captures":{"x":-32}}
main.test_single_int64: |
    {"stack_frames":[670512,672224,298500,517188],"captures":{"x":-64}}
main.test_single_rune: |
    {"stack_frames":[670416,672224,298500,517188],"captures":{"x":1}}
main.test_single_string:
main.test_single_uint: |
    {"stack_frames":[670528,672224,298500,517188],"captures":{"x":1512}}
main.test_single_uint8: |
    {"stack_frames":[670544,672224,298500,517188],"captures":{"x":8}}
main.test_single_uint16: |
    {"stack_frames":[670560,672224,298500,517188],"captures":{"x":16}}
main.test_single_uint32: |
    {"stack_frames":[670576,672224,298500,517188],"captures":{"x":32}}
main.test_single_uint64: |
    {"stack_frames":[670592,672224,298500,517188],"captures":{"x":64}}
main.test_slice_of_slices: ""
main.test_string_array: ""
main.test_string_pointer: ""
main.test_string_slice: ""
main.test_struct: ""
main.test_struct_slice: ""
main.test_three_strings: ""
main.test_three_strings_in_struct: ""
main.test_three_strings_in_struct_pointer: ""
main.test_type_alias: ""
main.test_uint_array: ""
main.test_uint_pointer: ""
main.test_uint_slice: ""
main.test_uint8_array: ""
main.test_uint16_array: ""
main.test_uint32_array: ""
main.test_uint64_array: ""
